### PR TITLE
[CSL-2144] Add system/shutdown endpoint

### PR DESCRIPTION
**Pull request by volhovm**
Same as #2273. Adds endpoint that triggers node shutdown.

### CODE REVIEW

File: wallet/src/Pos/Wallet/Web/Api.hs [hunk 1]
```diff
@@ -503,6 +506,21 @@ data WBackupApiRecord route = WBackupApiRecord
...
+--   /system
+-- ~~~~~~~~~~
+
+type WSystemApi = "system" :> ToServant (WSystemApiRecord AsApi)
+
+data WSystemApiRecord route = WSystemApiRecord
+  {
+    _requestShutdown :: route
+    :- "shutdown"
+    :> Summary "Import full information about wallet from a given file."
```
Comment (pva701) [inline wallet/src/Pos/Wallet/Web/Api.hs]: ahaha, it seems you copied this from import endpoint and forgot to rewrite, :lalka:

Comment (volhovm) [reply wallet/src/Pos/Wallet/Web/Api.hs]: Lol, nice.

Comment (gromakovsky) [reply wallet/src/Pos/Wallet/Web/Api.hs]: ahaha loh ppz mda :kaby:

File: wallet/src/Pos/Wallet/Web/Methods/Misc.hs [hunk 2]
```diff
@@ -113,6 +117,22 @@ postponeUpdate = removeNextUpdate >> return NoContent
...
+-- needed in order for http request to succeed.
+requestShutdown ::
+       ( MonadIO m
+       , MonadReader ctx m
+       , WithLogger m
+       , HasShutdownContext ctx
+       , Mockables m [Async, Delay]
+       )
+    => m NoContent
+requestShutdown = NoContent <$ async (delay (sec 1) >> triggerShutdown)
```
Comment (avieth) [inline wallet/src/Pos/Wallet/Web/Methods/Misc.hs]: This is dodgy.

Why does `triggerShutdown` cause the web server to fail to send a response?

**Pull request merged**